### PR TITLE
ruby version bump

### PR DESCRIPTION
**Pull request by imaia**
GOSU and TINI were outdated and received an update.

### CODE REVIEW

**Comment by tianon** [comment]
Looking at https://www.redmine.org/projects/redmine/wiki/redmineinstall#Ruby-interpreter, only Redmine 3.3 supports Ruby 2.3, so this will make our template (which isn't updated here) more complicated.  I'm not completely opposed, but I'd rather make the switch with a recommendation from upstream (because upstream clearly still supports 2.2, and I can't find any official recommendation one way or another which version they actively prefer for a given release). :confused:

(As a side note, if we move forward with this PR, we'll want to see the `.editorconfig` removed and the extraneous whitespace changes reverted.)

**Comment by imaia** [comment]
I will remove the .editorconfig and whitespaces by this week.

**Comment by tianon** [comment]
Well, before you do that (unless you _really_ want to do so), we need to figure out a good solution to updating the template, or this change will get overwritten by the updater bot, but even more importantly, we need to figure out whether this change is something we really want to commit to (preferably based on a recommendation from upstream, if we can find it).

File: 3.3/docker-entrypoint.sh [hunk 1]
```diff
@@ -106,17 +106,21 @@ case "$1" in
 		if [ "$1" != 'rake' -a -z "$REDMINE_NO_DB_MIGRATE" ]; then
 			gosu redmine rake db:migrate
 		fi
-		
+
 		chown -R redmine:redmine files log public/plugin_assets
-		
+
+    if [ "$1" != 'rake' -a -n "$REDMINE_PLUGINS_MIGRATE" ]; then
```
Comment (imaia) [inline 3.3/docker-entrypoint.sh]: These lines migrate the available plugins. Code by @HaMster21

**Comment by imaia** [comment]
Humm, so https://github.com/docker-library/redmine/blob/master/update.sh would have to be updated, which is not the actual files I'm updating here. I didn't know this was how the magic was rolling. I'll unrust my bash scripting skills and try a PR for it. Maybe it would also be nice to keep this PR open until then. 

And, by the way, **I really want to do it**, for two reasons: if it is not accepted, I'll learn how to do it. If it is accepted, plugins, which are very important for redmine, will be easier to handle. AND, this [2fa plugin](https://github.com/centosadmin/redmine_2fa) will also be usable with the default redmine docker image, as it requires ruby2.3 (which is compatible with ruby2.2).

**Comment by tianon** [comment]
Indeed, thanks. :+1:

**Pull request closed without merging**